### PR TITLE
Change locales file generation to use JSON sources

### PR DESCRIPTION
**Pull request by ClearlyClaire**
## Translation files and inheritance

Before this PR, glitch-soc translation files were JS files manually inheriting from vanilla's JSON files.

This PR changes locales file generation to use JSON sources, and still allow setting locale inheritance per-flavour by using a new `inherit_locales` in the flavours' `theme.yml` file.

The JSON files were obtained by doing:
```sh
sed -i -z "s/import inherited from '.*';\s*\nconst messages = //" *.js
sed -i "s/\s*\/\/.*//" *.js 
sed -i -z "s/;\s*export default .*/\n/" *.js
for i in *.js; do
  json5 $i | json_pp > ${i}on;
done
```

### Tooling

Change `yarn manage:translations` to ignore translations already used upstream (this is very hacky but I could not find a better way at the moment).

### CODE REVIEW

File: app/javascript/flavours/glitch/locales/en.json [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  "web_app_crash.change_your_settings": "Change your {settings}",
+  "web_app_crash.content": "You could try any of the following:",
+  "web_app_crash.debug_info": "Debug information",
+  "web_app_crash.disable_addons": "Disable browser add-ons or built-in translation tools",
+  "web_app_crash.issue_tracker": "issue tracker",
+  "web_app_crash.reload": "Reload",
+  "web_app_crash.reload_page": "{reload} the current page",
+  "web_app_crash.report_issue": "Report a bug in the {issuetracker}",
+  "web_app_crash.settings": "settings",
+  "web_app_crash.title": "We're sorry, but something went wrong with the Mastodon app."
```
Comment (Plastikmensch) [inline app/javascript/flavours/glitch/locales/en.json:219]: Should these be in glitch?

Comment (ClearlyClaire) [reply app/javascript/flavours/glitch/locales/en.json:219]: Yes, we have a different error boundary design

Comment (Plastikmensch) [reply app/javascript/flavours/glitch/locales/en.json:219]: Oh, okay, I didn't know that.

**Comment by Plastikmensch** [comment]
Hmmh I have an issue with `yarn manage:translations` downstream.
It also adds strings which only exist in a stash.
Maybe that I used `git stash push --staged` has something to do with this, but either way, this isn't desired.

**Comment by ClearlyClaire** [comment]
It's because you should clear the `build` directory before rebuilding and running `yarn manage:translations`

**Comment by Plastikmensch** [comment]
> It's because you should clear the `build` directory before rebuilding and running `yarn manage:translations`

Oh, I see, thanks!

**Pull request merged**